### PR TITLE
Enable lmdb for ConsensusState storage

### PR DESCRIPTION
**Pull request by rranjan3**
Remove in-memory hashmap. Enable lmdb storage to store
and retrieve ConsensusState data.

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

### CODE REVIEW

File: src/core/src/consensus_state.rs [hunk 1]
```diff
@@ -52,14 +51,14 @@ pub struct PopulationSample {
 *     block
 *
 */
-#[derive(Clone, Debug, PartialEq, Default)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default)]
 pub struct EstimateInfo {
     pub population_estimate : f64,
-    pub previous_block_id: BlockId,
+    pub previous_block_id: String,//BlockId,
```
Comment (askmish) [inline src/core/src/consensus_state.rs]: Add context for  BlockId comment

File: src/core/src/engine.rs [hunk 2]
```diff
@@ -29,10 +29,13 @@ use serde_json;
...
 use std::time::Instant;
 use enclave_sim::*;
-use std::collections::{HashMap};
+use std::collections::HashMap;
 use consensus_state::*;
-use consensus_state_store::{ConsensusStateStore, InMemoryConsensusStateStore};
+use consensus_state_store::ConsensusStateStore;
 use poet2_util;
+use database::config;
+use database::lmdb;
```
Comment (askmish) [inline src/core/src/engine.rs]: Please consider refactoring lmdb instance into consensus state in next iteration.

File: src/core/src/engine.rs [hunk 3]
```diff
@@ -504,6 +508,23 @@ fn validator_is_claiming_too_early( service: &mut Poet2Service )->bool
...
 
+fn create_context() -> Result<lmdb::LmdbContext, CliError> {
+    let path_config = config::get_path_config();
+    let statestore_path = &path_config.data_dir.join(config::get_filename());
+
+    lmdb::LmdbContext::new(statestore_path, 1, None)
+        .map_err(|err| CliError::EnvironmentError(format!("{}", err)))
+}
+
+fn open_statestore(ctx: &lmdb::LmdbContext) -> Result<ConsensusStateStore, CliError> {
```
Comment (askmish) [inline src/core/src/engine.rs]: How do we close state store and release the context created.

Comment (rranjan3) [reply src/core/src/engine.rs]: Will scope in context cleanup if need in later iteration.

File: src/core/src/engine.rs (hunk 2 above)
Comment (rranjan3) [reply src/core/src/engine.rs]: yes

File: src/core/src/consensus_state.rs (hunk 1 above)
Comment (rranjan3) [reply src/core/src/consensus_state.rs]: Updated context for both the fields BlockId and PeerId

**Comment by rranjan3** [comment]
Updated

**Pull request merged**